pkg/autoscaler/aggregation/bucketing_test.go

```diff
@@ -57,156 +59,259 @@ func TestTimedFloat64Buckets(t *testing.T) {
 		granularity: 5 * time.Second,
 		stats: []args{
 			{trunc5, pod, 1.0},
-			{trunc5.Add(3 * time.Second), pod, 1.0}, // same bucket
-			{trunc5.Add(6 * time.Second), pod, 1.0}, // next bucket
+			{trunc5.Add(3 * time.Second), pod, 11.0}, // same bucket
+			{trunc5.Add(6 * time.Second), pod, 1.0},  // next bucket
 		},
 		want: map[time.Time]float64{
-			trunc5:                      2.0,
+			trunc5:                      12.0,
 			trunc5.Add(5 * time.Second): 1.0,
 		},
 	}, {
-		name:  "empty",
-		stats: []args{},
-		want:  map[time.Time]float64{},
+		name:        "empty",
+		granularity: time.Second,
+		stats:       []args{},
+		want:        map[time.Time]float64{},
 	}}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			buckets := NewTimedFloat64Buckets(tt.granularity)
+			// New implementation test.
+			buckets := NewTimedFloat64Buckets(2*time.Minute, tt.granularity)
 			for _, stat := range tt.stats {
 				buckets.Record(stat.time, stat.name, stat.value)
 			}
 
 			got := make(map[time.Time]float64)
-			for time, bucket := range buckets.buckets {
-				got[time] = bucket
-			}
+			// Less time in future than our window is (2mins above), but more than any of the tests report.
+			buckets.ForEachBucket(trunc1.Add(time.Minute), func(t time.Time, b float64) {
+				// Since we're storing 0s when there's no data, we need to exclude those
+				// for this test.
+				if b > 0 {
+					got[t] = b
+				}
+			})
 
 			if !cmp.Equal(tt.want, got) {
 				t.Errorf("Unexpected values (-want +got): %v", cmp.Diff(tt.want, got))
 			}
-			if len(tt.want) == 0 && !buckets.isEmpty() {
-				t.Error("IsEmpty() = false, want true")
-			}
 		})
 	}
 }
 
+func TestTimedFloat64BucketsManyReps(t *testing.T) {
+	granularity := time.Second
+	trunc1 := time.Now().Truncate(granularity)
+	buckets := NewTimedFloat64Buckets(time.Minute, granularity)
+	for p := 0; p < 5; p++ {
+		trunc1 = trunc1.Add(granularity)
+		for t := 0; t < 5; t++ {
+			buckets.Record(trunc1, pod+strconv.Itoa(p), float64(p+t))
+		}
+	}
+	// So the buckets are:
+	// [0, 1, 2, 3, 4] = 10
+	// [1, 2, 3, 4, 5] = 15
+	// ...						 = 20, 25
```

Comment:
What happened here? (Formatting is off)

Reply:
Eh, I wanted to list the values, but just deleted.